src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/util/InterventionFactory.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util
+
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.DynamicFrameworkContract
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Intervention
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+
+class InterventionFactory(em: TestEntityManager) : EntityFactory(em) {
+  private val serviceCategoryFactory = ServiceCategoryFactory(em)
+  private val serviceProviderFactory = ServiceProviderFactory(em)
+  private val npsRegionFactory = NPSRegionFactory(em)
+  private val pccRegionFactory = PCCRegionFactory(em)
+
+  fun create(
+    id: UUID? = null,
+    createdAt: OffsetDateTime? = null,
+    title: String = "Sheffield Housing Services",
+    description: String = "Inclusive housing for South Yorkshire",
+  ): Intervention {
+    // for now we aren't allowing any of the contract details to be configured
```

Comment:
Just out of curiosity, howcome? If we want to create intervention test data, does this mean we can't customise the nps/pcc regions at creation? 
Does this also mean we will have both a nps and pcc region will be created/included in the contract when the InterventionFactory is used to create intervention data.

Reply:
no reason, more like just " this is still to-do "

also, if i actually call this method i get some weird spring persistence errors currently, so it needs more work. just wanted to get this merged to build on.